pkg/providers/imagefamily/nodeimage_test.go

```diff
@@ -171,60 +184,159 @@ var _ = Describe("NodeImageProvider tests", func() {
 			ctx = options.ToContext(ctx, testOptions)
 		})
 
-		It("should match expected images for Ubuntu2204", func() {
-			foundImages, err := nodeImageProvider.List(ctx, nodeClass)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(foundImages).To(ContainElements(getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, sigImageVersion, kubernetesVersion)))
-		})
-
-		// This test changes depending on the Kubernetes version, in effect making the following version-specific tests unnecessary.
-		// They are still kept for clarity and to ensure that the behavior is explicitly tested.
-		It("should match expected images for AzureLinux, depending on the Kubernetes version", func() {
-			nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.AzureLinuxImageFamily)
+		Context("List FIPS Images When FIPSMode Is Explicitly FIPS", func() {
+			BeforeEach(func() {
+				nodeClass.Spec.FIPSMode = lo.ToPtr(v1beta1.FIPSEnabled)
+			})
+
+			It("should match expected images for FIPS when using generic Ubuntu", func() {
+				nodeClass.Spec.ImageFamily = lo.ToPtr(v1beta1.UbuntuImageFamily)
+				foundImages, err := nodeImageProvider.List(ctx, nodeClass)
+				Expect(err).ToNot(HaveOccurred())
+				expectedImages := getExpectedTestSIGImages(*nodeClass.Spec.ImageFamily, v1beta1.FIPSEnabled, sigImageVersion, kubernetesVersion)
+				// equal functions as a checker for both content (including length) and order of image sets
```

Comment:
minor, style: I don't think we need a comment for what Equal function does - IMO it's implied with two lists that the Equal function asserts the lists must be exactly equal (element-wise equality, incl length)

Reply:
Applies throughout.